travis: add release support

hunchback opened the pull request:
I first go at auto creating a github release based on the sequence proposed here: https://prof.bagneris.net/posts/2017/github-releases-travis-ci-en.html

for now on creation of a tag I expect github to generate the release which would include:
- README.md
- the-mesh-for-data.src.tag.gz

hunchback commented:
addressing https://github.com/IBM/the-mesh-for-data/issues/120

ronenkat commented on .travis.yml:
```diff
@@ -136,3 +136,18 @@ jobs:
+deploy:
+  provider: releases
+  api_key:
+    secure: Es3z5bIx3N2en+vn6ogPyftwMoyUYE60s29gfBmThBd0UoQKKltGn5bCo1UnR5NpmNAyDlJtz2yN239IImp2RYwLGVXMrT/lqsTtsrEm8rKEgMWhteBdxWeZhBWMEip0PjZznO0gY9ALAH0IALvYqtVmo93ZtvXXbyYgzXleqNSGE1ChK3u7gFrNeEHKH1ivQ0xKozg3W4kaKAxwnbsi7KSxyb0o/aK8QGPya8Ldfn0l7DhxMuSe9a3l30dnoh0EOTEsH9yosYmhqG+e4erKBSdjKbzQpxzh9xpKHbw998p84OsVrSR8UyRNmzG6IhfISVRUsn73CtpG7imeMhTaezcXQ0Hy1r6WvumoSl7oOFLUz1b3R59bfR1XCYKLcWgK4ke ...
```
Should we have an api_key her? what is it for?

hunchback replied:
this key was created by the `travis` command line its' encrypted so that only travis service (which has the private key of travis) can decrypt and then use the api_key for creating the release ... so there is no security risk here.

roee88 replied:
@ronenkat are we good here?

roee88 reviewed (changes requested):
I don't understand what you are releasing. As far as I know, the source code in both zip and tar.gz format is attached to *any* github release already. What we need for the main repo is described in https://github.com/IBM/the-mesh-for-data/issues/82, including the first step which is to create an empty release but to tag docker images appropriately.

hunchback commented:
> I don't understand what you are releasing. As far as I know, the source code in both zip and tar.gz format is attached to _any_ github release already. 

I removed the tag.gz (although was not able to find that the tag.gz and zip are created _also_ in the case of a customized release (once we run this change we will know for sure). 

I added README.md just as an example / place holder.

> What we need for the main repo is described in #82, including the first step which is to create an empty release but to tag docker images appropriately.

my change causes the creation of a release automatically on tagging ... as opposed to manually invoking github UI, as our release include non-default files (such as change-log or other files) its' important the process is automatic.

As to tagging docker images, this currently has no meaning and can't be tested as we don't yet support redirecting all images in the project to use a specified tag other than `latest`, this will be handled after changes by @froesef and #42 and as these images (and helm charts) will be tagged in the public repos this change will not effect the format of the github release (which is the focus of this change).
